src/Accounts/Authentication/Factories/SshCredentialFactory.cs

```diff
@@ -69,7 +69,7 @@ public SshCredential GetSshCredential(IAzureContext context, RSAParameters rsaKe
                 throw new NullReferenceException(Resources.AuthenticationClientFactoryNotRegistered);
             }
 
-            var publicClient = tokenCacheProvider.CreatePublicClient();
+            var publicClient = tokenCacheProvider.CreatePublicClient(context.Environment.ActiveDirectoryAuthority, context.Account.GetCommonTenant());
```

Comment:
Common tenant may not be the same as the tenant that VM is in.

Reply:
Done